test/lit/passes/gsi-desc.wast

```diff
@@ -0,0 +1,188 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py --all-items and should not be edited.
+;; RUN: foreach %s %t wasm-opt --gsi -all --closed-world -S -o - | filecheck %s
+
+(module
+  (rec
+    ;; CHECK:      (rec
+    ;; CHECK-NEXT:  (type $super (sub (descriptor $super.desc (struct))))
+    (type $super (sub (descriptor $super.desc (struct))))
+    ;; CHECK:       (type $super.desc (sub (describes $super (struct))))
+    (type $super.desc (sub (describes $super (struct))))
+
+    ;; CHECK:       (type $A (sub $super (descriptor $A.desc (struct))))
+    (type $A (sub $super (descriptor $A.desc (struct))))
+    ;; CHECK:       (type $A.desc (sub $super.desc (describes $A (struct))))
+    (type $A.desc (sub $super.desc (describes $A (struct))))
+
+    ;; CHECK:       (type $B (sub $super (descriptor $B.desc (struct))))
+    (type $B (sub $super (descriptor $B.desc (struct))))
+    ;; CHECK:       (type $B.desc (sub $super.desc (describes $B (struct))))
+    (type $B.desc (sub $super.desc (describes $B (struct))))
+
+    ;; CHECK:       (type $two (sub (descriptor $two.desc (struct))))
+    (type $two (sub (descriptor $two.desc (struct))))
+    ;; CHECK:       (type $two.desc (sub (describes $two (struct))))
+    (type $two.desc (sub (describes $two (struct))))
+
+    ;; CHECK:       (type $three (sub (descriptor $three.desc (struct))))
+    (type $three (sub (descriptor $three.desc (struct))))
+    ;; CHECK:       (type $three.desc (sub (describes $three (struct))))
+    (type $three.desc (sub (describes $three (struct))))
+  )
+
+  ;; CHECK:      (type $10 (func (param (ref $super) (ref $A) (ref $B))))
+
+  ;; CHECK:      (type $11 (func (param (ref $two) (ref $three))))
+
+  ;; CHECK:      (global $A.unnested.4294967295 (ref (exact $A.desc)) (struct.new_default $A.desc))
+
+  ;; CHECK:      (global $A (ref $A) (struct.new_default $A
+  ;; CHECK-NEXT:  (global.get $A.unnested.4294967295)
+  ;; CHECK-NEXT: ))
+  (global $A (ref $A) (struct.new $A
+    (struct.new $A.desc)
+  ))
+
+  ;; CHECK:      (global $B.unnested.4294967295 (ref (exact $B.desc)) (struct.new_default $B.desc))
+
+  ;; CHECK:      (global $two.unnested.4294967295 (ref (exact $two.desc)) (struct.new_default $two.desc))
+
+  ;; CHECK:      (global $two2.unnested.4294967295 (ref (exact $two.desc)) (struct.new_default $two.desc))
```

Comment:
Can we update the name to use "desc" instead of this very large number?

Reply:
Sure, done.